### ISSUE TITLE
Add static types in codebase

### ISSUE THREAD

**Issue by iamareebjamal** [fossasia/open-event-server#6463]
A lot of benefits can be reaped and a log of bugs can be caught with static typing in Python using validators and linters like mypy

This is a parent issue tracking various areas where static typing can be added and is reserved for first time and new contributors

https://docs.python.org/3/library/typing.html
https://mypy.readthedocs.io/en/stable/index.html

This issue can be broken down into 2 phases

1. Add typing annotations

Add annotations in:
- [ ] Utilities
- [ ] API helpers
- [ ] SQLAlchemy Models https://github.com/dropbox/sqlalchemy-stubs
- [ ] More to come

2. Test using mypy

Once enough annotations are added and not necessarily all the points in above step are completed, we want to

- [ ] Add mypy test in CI to ensure types are passed correctly

**Comment by Dilshaad21** [comment]
@iamareebjamal can I take this issue

**Comment by iamareebjamal** [comment]
Take one part of it. Analyze the codebase and tell me which component you'd like to work on

**Comment by Dilshaad21** [comment]
I would like to work with the utils component. Then how should I begin with it?

**Comment by sameshl** [comment]
@iamareebjamal I would like to take the api helpers part

**Comment by iamareebjamal** [comment]
@Dilshaad21 There are a lot of utilities in api.helpers as well unfortunately

Like this test - https://github.com/fossasia/open-event-server/blob/development/tests/all/unit/api/helpers/test_utilities.py

It is testing things from helpers but they are actually utilities.

Start with common.py and take up utilities in helpers. Anything not related to API is a utility - file utility, compression, text transforms etc

API helpers are directly related to the API, like discount code validation, etc which @sameshl can work on

**Comment by sargamm** [comment]
@iamareebjamal  Can I take this issue for the API handlers part? I was planning to start with db.py

**Comment by chiral-carbon** [comment]
@prateekj117 @iamareebjamal can I work the SQLAlchemy models part?

**Comment by prateekj117** [comment]
@chiral-carbon Ok, go ahead.

**Comment by himanshupathak21061998** [comment]
@chiral-carbon If you are not working on this I want to work on this issue.
@prateekj117 I will start by adding annotation and after that, I will add mypy test

**Comment by chiral-carbon** [comment]
@himanshupathak21061998 hey, i'm still working on this.

**Comment by iamareebjamal** [comment]
@chiral-carbon 26 days have passed without any progress. Besides, this issue is huge, there is no way 2 or even 5 contributors will have conflicts or work on same area when adding static types. So, not just 1 contributor can solve this issue

**Comment by chiral-carbon** [comment]
@iamareebjamal  okay.

**Comment by dinolinjob** [comment]
I would like to work on this issue @iamareebjamal

**Comment by iamareebjamal** [comment]
OK

**Comment by dinolinjob** [comment]
@iamareebjamal Is there a way to check whether the typing annotations written is correct before writing mypy test?
I have began adding typing annotations to auth.py in helpers. Any suggestions?

**Comment by iamareebjamal** [comment]
No such way

**Comment by HardikM1999** [comment]
Can I work on this issue @iamareebjamal ?

**Comment by iamareebjamal** [comment]
Yes, one more to the list

**Comment by leson238** [comment]
Can I work on this @iamareebjamal ?

**Comment by EkanshMangal** [comment]
Hello Sir, Can I work on this issue?

**Comment by snitin315** [comment]
@EkanshMangal Yes, you can.

**Comment by iamareebjamal** [comment]
Yes, you can like the others who mentioned above and never did. So please, instead of spamming "Can I work on this issue?" on every issue, which is very annoying, pick one and send a PR, which by judging previous behaviours of commentators won't happen

**Comment by b-thebest** [comment]
Hello, Can I work on this issue.

**Comment by iamareebjamal** [comment]
Yes, please go ahead

**Comment by imrahul361** [comment]
Is anyone working on this issue?

**Comment by iamareebjamal** [comment]
Considering it has been 6 months, no

**Comment by hari193023** [comment]
can I work on this issue?

**Comment by Shakil-786** [comment]
can I work on this?
